test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java

```diff
@@ -626,32 +628,49 @@ public String toString() {
         return result;
     }
 
+    private static boolean matchWithRange(String nodeVersionSting, List<VersionRange> acceptedVersionRanges) {
+        try {
+            Version version = Version.fromString(nodeVersionSting.replace("-SNAPSHOT", ""));
+            return acceptedVersionRanges.stream().anyMatch(v -> v.contains(version));
+        } catch (IllegalArgumentException ignored) {}
```

Comment:
I don't think we should be swallowing exceptions. If a version comes back malformed, the test should fail?

Reply:
Same as above: avoid failing the test is we are testing a serverless cluster. Or any node with a "opaque"/non-semantic version really.
In this case, "current" will still work as usual, and a string like "abcd" will simply not match a range of versions (like 8.1.0 - 8.12.0) - see the corresponding unit tests.

I can add a comment in the catch block to make this clearer.

Reply:
Or even better, I will do it as we do in other places and use a regex to detect semantic versions - which will make the code more intentional.

Reply:
After additional thoughts, you are very right here: it does not make sense to specify a version range for tests on a cluster with opaque version strings.
The exception is now propagated, with a useful error message to indicate where the problem is. See the spun off PR https://github.com/elastic/elasticsearch/pull/100794